CONTRIBUTING.md

```diff
@@ -3,41 +3,181 @@
+The goal is sustained 80% coverage and higher.
+Test files must have 100% coverage of themselves.
+Tests should generally be grouped into a file that is parallel in location
+under the `tests` directory to the file containing the code being tested.
+For example, if new code was added to the file `smqtk_core/utils/foo.py`, tests
+for that code should be added to `tests/utils/test_foo.py`.
+
+The `pytest` package is utilized as the test runner.
+See `pytest` documentation for further best-practices in writing and formatting
+tests.
+
+### Release Notes
+When making a topic branch, included in its commits should be additions to the
+[pending release notes] file.
+These additions should be short, descriptive summaries of the updates,
+features or fixes that have been added in the form of bullet points under
+applicable sections.
+There are broad sections for "Updates" (features or enhancements) and "Fixes"
+(typos, bug fixes, etc.).
+Most changes based on a ``release`` branch go under "Fixes", and most changes
+based on the ``master`` branch go under "Updates".
+
+A topic branch is generally required to provide at least one update to this
+file for merger approval.
+
+#### Release Note Contribution Exception
+When a new contribution is fixing a bug or minor issue with something that has
+an existing release note, it may be the case that no additional release notes
+are needed since they would add redundancy to the document.
+
+For example, if the PR is to fix a bug on the master branch that was introduced
+since the last release, and the fix does not impact how the feature is
+summarized in the release notes, then this should **not** be documented in the
+"Fixes" section of the release notes because the bug itself was never released.
+In such a case, the release-notes CI check on that follow-on contribution may
+be ignored by the reviewers of the contribution.
+
+Generally, a reviewer will assume that a release note is required unless the
+contributor makes a case that the check should be ignored.
+This will be considered by reviewers on a case-by-case basis.
+
+### Continuous Integration
+SMQTK-Core uses GitHub for code review and GitHub Actions for continuous
+testing as new pull requests are made.
+Most, but not all, CI checks are required to pass before a PR can be merged.
+It is generally recommended that *all* checks pass, but there may be exception
+cases for non-required CI checks.
+
+See the [review process documentation] for more details on the CI checks
+utilized.
+
+### Code Reviews
+Pull requests are reviewed by one or more of the core SMQTK-Core maintainers
+using the GitHub tools for discussions.
+Maintainers should not merge a PR until it conforms to the requirements
+described here (e.g. coding style, release notes, etc.) and it is confirmed
+that the code has sufficient unit tests and does not break any existing unit
+tests.
+
+The [review process documentation] is described in more details.
```

Comment:
This sentence sounds a little awkward to me. Maybe could do something like:
"See the [review process documentation] for more details on the code review process."

Reply:
Good catch! That sentence was likely the result of an unfinished thought. I'll get an update here in a sec.